python/tk_framework_desktopserver/shotgun/api_v1.py

```diff
@@ -26,11 +26,21 @@ class ShotgunAPI():
     Public API
     Callable methods from client. Every one of these methods can be called from the client.
     """
-    public_api = ["echo", "open", "executeToolkitCommand", "executeTankCommand",
-                  "pickFileOrDirectory", "pickFilesOrDirectories", "version",
-                  "getProjectActions"]
-
-    def __init__(self, host, process_manager):
+    PUBLIC_API_METHODS = [
+        "echo",
+        "open",
+        "executeToolkitCommand",
+        "executeTankCommand",
+        "pickFileOrDirectory",
+        "pickFilesOrDirectories",
+        "version",
+        "getProjectActions",
+    ]
+
+    # Nothing for v1. This is used in v2.
+    SYNCHRONOUS_METHODS = []
```

Comment:
The way that v1 works at the moment is that if you keep hitting refresh it will spin up more and more and more `tank`s that will all read the config. Doesn't need to be fixed right away, but might be nice to turn on if we think it could speed up the menu generation.